onmt/modules/Sequencer.lua

```diff
@@ -24,7 +24,25 @@ function Sequencer:__init(network)
 end
 
 function Sequencer:_sharedClone()
-  local clone = self.network:clone('weight', 'gradWeight', 'bias', 'gradBias', 'fullWeight', 'fullBias')
+  local clone = self.network:clone('weight', 'gradWeight', 'bias', 'gradBias', 'fullWeight', 'fullBias', 'noiseInit')
+
+  -- for tensors/table that can change dimension - copy the tensor itself
+  local copyBuffers = { 'sharedNoise' }
+  local buffersToCopy = {}
+  self.network:apply(function(m)
```

Comment:
We could rely on the graph traversal that occurs just below, for the Tensor sharing.